### PR TITLE
Add CFPropertyList serialization

### PR DESCRIPTION
**Pull request by cjhowe**
Tests included. Reopened from #88

---
This change is [Reviewable](https://reviewable.io/reviews/servo/core-foundation-rs/90)

### CODE REVIEW

File: core-foundation/src/propertylist.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+use data::CFData;
+use base::{CFType, TCFType};
+
+pub use core_foundation_sys::propertylist::*;
+use core_foundation_sys::error::CFErrorRef;
+use core_foundation_sys::base::{CFTypeRef, kCFAllocatorDefault};
+
+pub fn create_with_data(data: CFData,
+                        options: CFPropertyListMutabilityOptions)
+                        -> Result<(CFType, CFPropertyListFormat), CFError> {
```
Comment (cjhowe) [inline core-foundation/src/propertylist.rs]: I'm not sure that using CFType here is appropriate. Other parts of the API use `*const c_void` for these generic types instead.

Comment (cjhowe) [inline core-foundation/src/propertylist.rs]: I've now converted this to use `*const c_void`, keeping it consistent with other parts of the API.

File: core-foundation/src/propertylist.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    unsafe {
+        let mut error: CFErrorRef = ptr::null_mut();
+        let mut format: CFPropertyListFormat = 0;
+        let property_list = CFPropertyListCreateWithData(kCFAllocatorDefault,
+                                                         data.as_concrete_TypeRef(),
+                                                         options,
+                                                         &mut format as *mut CFPropertyListFormat,
+                                                         &mut error as *mut CFErrorRef);
```
Comment (jdm) [inline core-foundation/src/propertylist.rs]: These casts shouldn't be necessary.

File: core-foundation/src/propertylist.rs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+pub fn create_data(property_list: CFType, format: CFPropertyListFormat) -> Result<CFData, CFError> {
+    unsafe {
+        let mut error: CFErrorRef = ptr::null_mut();
+        let property_list_ref: CFTypeRef = property_list.as_CFTypeRef();
+        let data_ref = CFPropertyListCreateData(kCFAllocatorDefault,
+                                                property_list_ref,
+                                                format,
+                                                0,
+                                                &mut error as *mut CFErrorRef);
```
Comment (jdm) [inline core-foundation/src/propertylist.rs]: This cast shouldn't be necessary.

**Comment by cjhowe** [comment]
Unnecessary casts removed. Thanks for the reviews!

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Pull request merged**